### PR TITLE
Support Winsock in crustls-demo

### PR DESCRIPTION
**Pull request by gvanem**
As per the issue https://github.com/abetterinternet/crustls/issues/15, here is my feeble attempt at this demo.
Tested with MSVC / clang-cl only

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -30,23 +61,45 @@ print_error(char *prefix, rustls_result result)
...
+const char *ws_strerror (int err)
+{
+  static char ws_err[50];
+
+  if (err >= WSABASEERR) {
+    snprintf(ws_err, sizeof(ws_err), "Winsock err: %d", err);
+    return ws_err;
+  }
+  /* Assume a CRT error */
+  return (strerror)(err);
```
Comment (jsha) [inline src/main.c:74]: Why is strerror in parentheses here?

File: src/main.c [hunk 2]
```diff
@@ -30,23 +61,45 @@ print_error(char *prefix, rustls_result result)
...
+}
+#endif
+
 /*
  * Write n bytes from buf to the provided fd, retrying short writes until
  * we finish or hit an error. Assumes fd is blocking and therefore doesn't
  * handle EAGAIN. Returns 0 for success or 1 for error.
+ *
+ * For Windsock we cannot use a socket-fd in write().
```
Comment (jsha) [inline src/main.c:83]:
```suggestion
 * For Winsock we cannot use a socket-fd in write().
```

File: src/main.c [hunk 3]
```diff
@@ -30,23 +61,45 @@ print_error(char *prefix, rustls_result result)
...
+ * Call send() if fd > STDOUT_FILENO.
```
Comment (jsha) [inline src/main.c:84]: I think `send` and `recv` are supported on both Linux and macOS. I'd be happier splitting this function into two: `write_all` (for FDs) and `send_all` (for sockets).

**Review by jsha — CHANGES_REQUESTED** [review]
Awesome, thanks for writing this!

File: src/main.c (hunk 1 above)
Comment (gvanem) [reply src/main.c:74]: To get the original CRT `strerror()` back after I changed it.

File: src/main.c (hunk 3 above)
Comment (gvanem) [reply src/main.c:84]: That's what I was thinking too. But that would be too many diffs.

**Pull request merged**